src/test/socket/ioctl/test_ioctl.rs

```diff
@@ -303,7 +303,12 @@ fn test_siocgstamp(init_method: SocketInitMethod, sock_type: libc::c_int) -> Res
 
                 // since it was sent over localhost, the difference between the send time and
                 // receive time should be very small
-                test_utils::result_assert(difference < threshold, "Time difference was too large")?;
+                assert!(
```

Comment:
don't we want to return an error instead of panicking?

Reply:
Originally we did because we wanted the `--summarize` option, which required us to return an error instead of panicking. But in #3207 I think we want to remove this option, which means we can switch to panics instead of returning errors.

But for consistency, I added new `result_assert_lt` and `result_assert_gt` utils, and changed this back to returning a result.

Reply:
Ah ok. Yeah I'm in favor of simplifying things as per #3207, but makes sense to be  consistent for now.

I'd have settled for just sticking a (mildly wasteful) `format!` call into the `result_assert` call we had before, but the new helpers look good too :)

Reply:
I didn't even think of the `format!` macro... Yeah that would have been simpler